packages/three-vrm/src/VRMUtils/combineSkeletons.ts

```diff
@@ -9,98 +9,193 @@ import * as THREE from 'three';
  * @param root Root object that will be traversed
  */
 export function combineSkeletons(root: THREE.Object3D): void {
-  const skinnedMeshes = new Set<THREE.SkinnedMesh>();
-  const geometryToSkinnedMesh = new Map<THREE.BufferGeometry, THREE.SkinnedMesh>();
+  const skinnedMeshes = collectSkinnedMeshes(root);
+
+  // List all bones and boneInverses for each meshes
+  const attributeUsedIndexSetMap = new Map<THREE.BufferAttribute | THREE.InterleavedBufferAttribute, Set<number>>();
+  const meshBoneInverseMapMap = new Map<THREE.SkinnedMesh, Map<THREE.Bone, THREE.Matrix4>>();
+  for (const mesh of skinnedMeshes) {
+    const boneInverseMap = listUsedBones(mesh, attributeUsedIndexSetMap);
+    meshBoneInverseMapMap.set(mesh, boneInverseMap);
```

Comment:
I think the side effects are not clear. How about setting meshBoneInverseMapMap within the function?